Add "no_run_validators()" context manager (#859)
python-attrs/attrs: docs/api.rst

            ...
        TypeError: ("'x' must be <class 'str'> (got 7 that is a <class 'int'>).", Attribute(name='x', default=NOTHING, validator=<deep_mapping validator for objects mapping <instance_of validator for type <class 'str'>> to <instance_of validator for type <class 'int'>>>, repr=True, cmp=True, hash=None, init=True, metadata=mappingproxy({}), type=None, converter=None, kw_only=False), <class 'str'>, 7)

> Validators can be globally disabled if you want to run them only in development and tests but not in production because you fear their performance impact:
> 
> .. autofunction:: attr.validators.set_disabled
> 
> .. autofunction:: attr.validators.get_disabled
> 
> .. autofunction:: attr.validators.disabled
> 

Reviewer: I think we should push the contextmanager as the prime API:

```suggestion
Validators can be both globally and locally disabled:

.. autofunction:: attr.validators.disabled

.. autofunction:: attr.validators.set_disabled

.. autofunction:: attr.validators.get_disabled

```
